### DISCUSSION TITLE
Centralized logging to Log Analytics?

### DISCUSSION THREAD

**Discussion by mtengmo** [jantari/LSUClient#72] [Show and tell]
**Describe the solution you'd like**
Centralize logging to a Log Analytics workspace. 
**Describe alternatives you've considered**
Save to fileserver

**Additional context**
https://wintellisys.com/store-custom-logs-in-azure-log-analytics-workspace-using-powershell/

**Comment by jantari** [comment]
This is a very cool idea and certainly possible.

It is outside of the scope of LSUClient to integrate this as a built-in feature, but you can push any of the data you get back from LSUClient to Azure Monitor yourself with the API calls in the article you linked. Unfortunately I have not had the chance to do much with Azure Monitor yet myself so I personally cannot help you with that.

If you have a small working example, maybe we can add it to the [docs](https://jantari.github.io/LSUClient-docs/) which are also built from a [public repository](https://github.com/jantari/LSUClient-docs) here on GitHub. There is already a simple example of logging in the best-practices article: https://github.com/jantari/LSUClient-docs/blob/main/content/docs/topics/best-practices.md#3-write-a-logfile so this would fit in there I think.

**Comment by mtengmo** [comment]
I got a small start at least :) 
https://github.com/mtengmo/Public/blob/master/Powershell/LSUClient-Logging/Run-LSUClient.ps1

Not really sur how to get the write-* outputs to a json object so it could be sent to the log analytics workspace. 

![image](https://user-images.githubusercontent.com/17195619/212659387-0f9dd05c-8d43-46fc-8c8e-162a2543b6e3.png)

**Comment by mtengmo** [comment]
Updated the script above with more logging.

**Comment by jantari** [comment]
I've enabled the GitHub "Discussions" feature for this repository and moved this issue over here so it can stay open indefinitely, separate from actual issue reports, and everyone who's interested in this topic can kind of follow along or contribute how they're approaching this.
